src/coreclr/vm/stdinterfaces_wrapper.cpp

```diff
@@ -660,1753 +377,782 @@ HRESULT __stdcall ErrorInfo_GetDescription_Wrapper(IUnknown* pUnk, BSTR* pbstrDe
+            PRECONDITION(CheckPointer(pMarsh));
+        }
+        CONTRACTL_END;
+    
+        if (!CanCallRuntimeInterfaceImplementations())
+            return E_FAIL;
 
-VOID __stdcall Dispatch_GetTypeInfo_CallBack (LPVOID ptr)
-{
-    CONTRACTL
-    {
-        NOTHROW;
-        GC_TRIGGERS;
-        MODE_PREEMPTIVE;
-        PRECONDITION(CheckPointer(ptr));
+        return Marshal_DisconnectObject(pMarsh, dwReserved);
     }
-    CONTRACTL_END;
 
-    GetTypeInfoArgs* pArgs = (GetTypeInfoArgs*)ptr;
-    ComCallWrapper* pWrap = MapIUnknownToWrapper(pArgs->pUnk);
-    if (IsCurrentDomainValid(pWrap))
+    // ---------------------------------------------------------------------------
+    //  Interface IConnectionPointContainer
+
+    HRESULT __stdcall ConnectionPointContainer_EnumConnectionPoints_Wrapper(IUnknown* pUnk, IEnumConnectionPoints **ppEnum)
     {
-        *(pArgs->hr) = Dispatch_GetTypeInfo(pArgs->pUnk, pArgs->itinfo, pArgs->lcid, pArgs->pptinfo);
+        SetupForComCallHR();
+
+        CONTRACTL
+        {
+            NOTHROW;
+            GC_TRIGGERS;
+            MODE_PREEMPTIVE;
+            PRECONDITION(CheckPointer(pUnk));
+            PRECONDITION(CheckPointer(ppEnum, NULL_OK));
+        }
+        CONTRACTL_END;
+    
+        if (!CanCallRuntimeInterfaceImplementations())
+            return E_FAIL;
+
+        return ConnectionPointContainer_EnumConnectionPoints(pUnk, ppEnum);
     }
-    else
+
+    HRESULT __stdcall ConnectionPointContainer_FindConnectionPoint_Wrapper(IUnknown* pUnk, REFIID riid, IConnectionPoint **ppCP)
     {
-        AppDomainDoCallBack(pWrap, Dispatch_GetTypeInfo_CallBack, pArgs, pArgs->hr);
-    }
-}
+        SetupForComCallHR();
 
-HRESULT __stdcall Dispatch_GetTypeInfo_Wrapper(IDispatch* pDisp, unsigned int itinfo, LCID lcid, ITypeInfo **pptinfo)
-{
-    SetupForComCallHR();
+        CONTRACTL
+        {
+            NOTHROW;
+            GC_TRIGGERS;
+            MODE_PREEMPTIVE;
+            PRECONDITION(CheckPointer(pUnk));
+            PRECONDITION(CheckPointer(ppCP, NULL_OK));
+        }
+        CONTRACTL_END;
+    
+        if (!CanCallRuntimeInterfaceImplementations())
+            return E_FAIL;
 
-    CONTRACTL
-    {
-        NOTHROW;
-        GC_TRIGGERS;
-        MODE_PREEMPTIVE;
-        PRECONDITION(CheckPointer(pDisp));
-        PRECONDITION(CheckPointer(pptinfo, NULL_OK));
+        return ConnectionPointContainer_FindConnectionPoint(pUnk, riid, ppCP);
     }
-    CONTRACTL_END;
 
-    HRESULT hr = S_OK;
-    GetTypeInfoArgs args = {pDisp, itinfo, lcid, pptinfo, &hr};
-    Dispatch_GetTypeInfo_CallBack(&args);
-    return hr;
-}
 
-struct GetIDsOfNamesArgs
-{
-    IDispatch* pUnk;
-    const IID* riid;
-    OLECHAR **rgszNames;
-    unsigned int cNames;
-    LCID lcid;
-    DISPID *rgdispid;
-    HRESULT* hr;
-};
+    //------------------------------------------------------------------------------------------
+    //      IObjectSafety methods for COM+ objects
```

Comment:
I would be surprised if `IObjectSafety` is still a valid scenario. That is an old IE interface for ActiveX controls.

Reply:
It would technically be a breaking change to remove the automatic `IObjectSafety` implementation, so I'd prefer to do that in another PR if we want to do that.